devito/finite_differences/operators.py

```diff
@@ -0,0 +1,61 @@
+from devito.logger import error
+
+
+def div(func):
+    """
+    Divergence of the input func.
+    Func can be a Function, VectorFunction or TensorFunction
+    """
+    try:
+        return func.div
+    except AttributeError:
+        return 0
+
+
+def grad(func):
+    """
+    Gradient of the input func.
+    Func can be a Function or a VectorFunction
+    """
+    try:
+        return func.grad
+    except AttributeError:
+        error("Gradient not supported for class %s" % func.__class__)
+
+
+def curl(func):
+    """
+    Curl of the input func.
+    Func can be a Function or a VectorFunction
+    """
+    try:
+        return func.curl
+    except AttributeError:
+        error("Curl only supported for 3D VectorFunction and VectorTimeFunction")
+
+
+def Laplacian(func):
+    """
+    Laplacian of the input func.
+    Func can be a Function, VectorFunction or TensorFunction
+    """
+    try:
+        return func.laplace
+    except AttributeError:
+        return 0
+
+
+def diag(func, size=None):
+    """
+    Creates the diagonal Tensor with func on its diagonal
+    """
```

Comment:
Maybe add what functions can `Func` be here as well like Laplacian, curl, etc?

Reply:
Well it can be anything you want, can be a floating point value if you feel like it too.

Reply:
Can be a pure numpy number too

Reply:
how about: `Differentiable or scalar`

Reply:
fixed